samza-api/src/main/java/org/apache/samza/table/remote/TableRateLimiter.java

```diff
@@ -79,11 +75,12 @@
    * @param tag tag to be used with the rate limiter
    */
   public TableRateLimiter(String tableId, RateLimiter rateLimiter, CreditFunction<K, V> creditFn, String tag) {
+    Preconditions.checkNotNull(rateLimiter);
```

Comment:
should tableId and creditFn be null checked as well?

Reply:
Good catch, creditFn is optional though.